clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java

```diff
@@ -0,0 +1,303 @@
+   * @param recursive if true, the parent directories to the file in Tachyon will be created
+   * @return the file id of the resulting file in Tachyon
+   * @throws IOException if the Tachyon path is invalid or the ufsPath does not exist
+   */
+  public long loadFileFromUfs(TachyonURI path, TachyonURI ufsPath, boolean recursive)
+      throws IOException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      return masterClient.loadFileFromUfs(path.getPath(), ufsPath.getPath(), -1L, recursive);
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Creates a folder. If the parent folders do not exist, they will be created automatically.
+   *
+   * @param path the handler for the file
+   * @return true if the folder is created successfully or already existing, false otherwise.
+   * @throws IOException if the master cannot create the folder under the specified path
+   */
+  @Override
+  public boolean mkdirs(TachyonURI path) throws IOException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      // TODO: Change this RPC's arguments
+      return masterClient.createDirectory(path.getPath(), true);
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Resolves a {@link TachyonURI} to a {@link TachyonFile} which is used as the file handler for
+   * non-create operations.
+   *
+   * @param path the path of the file, this should be in Tachyon space
+   * @return a TachyonFile which acts as a file handler for the path
+   * @throws IOException if the path does not exist in Tachyon space
+   */
+  @Override
+  public TachyonFile open(TachyonURI path) throws IOException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      return new TachyonFile(masterClient.getFileId(path.getPath()));
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Sets the pin status of a file. A pinned file will never be evicted for any reason. The pin
+   * status is propagated asynchronously from this method call on the worker heartbeats.
+   *
+   * @param file the file handler for the file to pin
+   * @param pinned true to pin the file, false to unpin it
+   * @throws IOException if an error occurs during the pin operation
+   */
+  public void setPin(TachyonFile file, boolean pinned) throws IOException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      masterClient.setPinned(file.getFileId(), pinned);
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Renames an existing file in Tachyon space to another path in Tachyon space.
+   *
+   * @param src The file handler for the source file
+   * @param dst The path of the destination file, this path should not exist
```

Comment:
+1 to clarify this in the doc: the given path should not exist, but the parent path should